### PR TITLE
feat: First pass generating CSV reports of scan

### PR DESCRIPTION
**Pull request by QuentinMoss**
First pass adding support CSV report generation triggered by CLI arg.
https://github.com/resyncgg/armada/issues/10

### CODE REVIEW

File: armada/src/main.rs [hunk 1]
```diff
@@ -1,56 +1,88 @@
...
+            .run_with_stats(targets, ports, source_ipv4, source_ipv6, retries, timeout, rate_limit, stream_results, generate_report)
             .await
     };
 
-    if !stream_results {
-        syn_scan_results.sort();
 
-        syn_scan_results.into_iter().for_each(|remote| {
+    if !stream_results {
+        let scan_results = &syn_scan_results;
```
Comment (QuentinMoss) [inline armada/src/main.rs:56]: @d0nutptr I'm not sure of the most performant way to do this. Ideally we would avoid unnecessary allocations, but into_iter() is causing issues with borrowing for anything else that may reference it later
 
I think cloning just in the scope of report generating should be ok though?

**Comment by d0nutptr** [comment]
Thank you so much for this change! 

I think what I, and others, want here is the ability to control what is being written to `stdout`. This change seems to take the approach of adding a `--report` flag which adds a separate report file being written. 

I think what this change calls for is instead the following:

* new, optional `--output` (`-o`) flag that sets the output format
* update the printing to change how it prints based on the output format flag.
* variants could just be `"csv"` and `"default"` for now

Would it be okay to make those changes first?

**Comment by QuentinMoss** [comment]
@d0nutptr ON IT!!!

**Comment by QuentinMoss** [comment]
2 confused getting this glued together. Closing

**Pull request closed without merging**